ts/Core/Chart/MapChart.ts

```diff
@@ -198,6 +216,242 @@ namespace MapChart {
+        levelToRemove: (number|undefined);
+
+    if (drilldownLevels && drilldownLevels.length > 0) { // #3352, async loading
+        levelToRemove = drilldownLevels[drilldownLevels.length - 1].levelNumber;
+        (this.drilldownLevels as any).forEach(function (
+            level: Highcharts.DrilldownLevelObject,
+            i: Number
+        ): void {
+
+            if (
+                chart.options.drilldown &&
+                chart.options.drilldown.mapZooming
+            ) {
+                chart.redraw();
+                if (chart.mapView) { // TO DO try to remove MapSeries
+                    chart.mapView.fitToBounds(
+                        (level.lowerSeries as MapSeries).bounds);
+                }
+            }
+
+            if (level.levelNumber === levelToRemove) {
+                level.levelSeries.forEach(function (series, j): void {
+                    // Not removed, not added as part of a multi-series
+                    // drilldown
+                    if (
+                        series.options &&
+                        series.options._levelNumber === levelToRemove &&
+                        series.group
+                    ) {
+                        let animOptions: boolean | Partial<AnimationOptions> | undefined = {};
+
+                        if (chart.options.drilldown) {
+                            animOptions = chart.options.drilldown.animation;
+                        }
+
+                        series.group.animate({
+                            opacity: 0
+                        },
+                        animOptions,
+                        function (): void {
+                            series.remove(false);
+
+                            // We have a reset zoom button. Hide it and detatch
+                            // it from the chart. It is preserved to the layer
+                            // config above.
+                            if (chart.resetZoomButton) {
+                                chart.resetZoomButton.hide();
+                                delete chart.resetZoomButton;
+                            }
+
+                            chart.pointer.reset();
+
+                            fireEvent(chart, 'afterDrilldown');
+
+                            if (chart.mapView) {
+                                chart.series.forEach((series): void => {
+                                    series.isDirtyData = true;
+                                });
+                                chart.mapView.setView(void 0, 1);
+                            }
+                            fireEvent(chart, 'afterApplyDrilldown');
+                        });
+                    }
+                });
+            }
+        });
+    }
+    return false; // to prevent default function from fireEvent
+});
+
+// to prevent default function from fireEvent
+addEvent(MapChart, 'midDrillUp', function (): boolean {
+    return !this.options.isMapChart;
```

Comment:
Check for `this instanceof MapChart` instead. Then we don't have to introduce the `isMapChart` property.